lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1311,7 +1362,8 @@ SPIRVValue *LLVMToSPIRV::transValueWithoutDecoration(Value *V,
   if (UnaryInstruction *U = dyn_cast<UnaryInstruction>(V)) {
     if (isSpecialTypeInitializer(U))
       return mapValue(V, transValue(U->getOperand(0), BB));
-    return mapValue(V, transUnaryInst(U, BB));
+    auto UI = transUnaryInst(U, BB);
+    return mapValue(V, UI ? UI : transValue(U->getOperand(0), BB));
```

Comment:
Should `transUnaryInst`  return `transValue(U->getOperand(0), BB)` instead of `nullptr`?

Reply:
`transUnaryInst` returns `SPIRVInstuction *`, while `transValue` returns `SPIRVValue *`, so it won't work.